### DISCUSSION TITLE
[Feature Request] upload multiple files in storage in one transaction

### DISCUSSION THREAD

**Discussion by zakaria-chahboun** [supabase/supabase#3147] [Feature Requests]
Hi everyone

I know how to upload a file by JavaScript API. But i have a lot of files, And i want to upload them all one time.
If there is an exception then just revoke the upload!

This question has a relation with this #3124 

Thanks :heart: !

**Comment by MildTomato** [comment]
Is the use case here that by combining all the files together, you are guaranteeing that all files are uploaded?

**Comment by inian** [comment]
Hi @zakaria-chahboun, this will be hard to implement in the middleware itself, since it wouldn't scale well if the number of files are large. It could take many minutes to ensure this "transaction" fails or completes. AFAIK Firebase or S3 also dont' support something similar either. 

I think the best way to do this is to handle this on the application layer where you can look at the status codes of the responses and delete the objects if needed. This is what our dashboard does as well, when you upload 1000s of files at the same time, we batch and upload to the storage server.
